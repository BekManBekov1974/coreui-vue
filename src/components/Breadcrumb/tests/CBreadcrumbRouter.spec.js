//TODO: the test doesn't set paths correctly in the snapshot
import { mount, createLocalVue } from '@vue/test-utils'
import Component from '../CBreadcrumbRouter'
import VueRouter from 'vue-router'

const localVue = createLocalVue()
const router = new VueRouter()

const ComponentName = 'CBreadcrumbRouter'
const wrapper = mount(
  Component,
  { 
    localVue, 
    router, 
    mocks: {
      $route: { 
        matched: [
          { path: '', name: 'Home', meta: {} },
          { path: '/menu', name: 'Menu', meta: {} },
          { path: '/menu/view', name: 'View', meta: {} } 
        ] 
      }
    },
    propsData: { 
      addLinkClasses: 'test-link-class',
      addLastItemClasses: 'test-last-item-class',
    }
  }
)


describe(ComponentName, () => {
  it('has a name', () => {
    expect(Component.name).toMatch(ComponentName)
  })
  it('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
});
