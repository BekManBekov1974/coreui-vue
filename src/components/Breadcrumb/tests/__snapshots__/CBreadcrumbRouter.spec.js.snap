// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CBreadcrumbRouter renders correctly 1`] = `
<ol
  class="c-breadcrumb"
>
  <li
    class="additional-class c-breadcrumb-item additional-link-class"
    role="presentation"
  >
    <a
      class=""
      href="#"
      target="_self"
    >
      Home
    </a>
  </li>
  <li
    class="additional-class c-breadcrumb-item additional-link-class"
    role="presentation"
  >
    <a
      class=""
      href="#"
      target="_self"
    >
      Menu
    </a>
  </li>
   
  <li
    class="c-active additional-class c-breadcrumb-item additional-last-item-class"
    role="presentation"
  >
    <span>
      View
    </span>
  </li>
</ol>
`;
